spec/mem/tree_borrows/memory.md

```diff
@@ -193,12 +218,17 @@ impl<T: Target> Memory for TreeBorrowsMemory<T> {
         allocation.deallocation_check(ptr.addr, kind, size, align)?;
 
         // Check that ptr has the permission to write the entire allocation.
-        allocation.extra.root.access(Some(path), AccessKind::Write, Size::ZERO, size)?;
+        allocation.extra.root.access(Some(path), AccessKind::Write, Offset::ZERO, size)?;
+
+        // Check that allocation is not strongly protected.
+        if allocation.extra.root.contain_strong_protector() {
```

Comment:
Alternatively to this, we could have an `AccessKind::Dealloc` and have `access` itself also check for strong protectors. That would avoid a second traversal over the full tree. I am honestly not sure what is better... @sun-jacobi  @JoJoDeveloping any opinion?

Reply:
This makes sense to me :)

Reply:
Okay, let's keep this as-is then. :)